### PR TITLE
daemon, cmd/snap: refresh --devmode

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: cmd/snap/cmd_snap_op.go [hunk 1]
```diff
@@ -285,6 +310,7 @@ func refreshAll() error {
 
 	names := make([]string, len(updates))
 	for i, update := range updates {
+		// TODO: set JailMode from snapstate
```
Comment (pedronis) [inline cmd/snap/cmd_snap_op.go]: but this doesn't belong here, no? it would need to be in snapstate or api itself

**Comment by pedronis** [comment]
seems to be missing some spread tests

File: cmd/snap/cmd_snap_op.go [hunk 2]
```diff
@@ -189,33 +189,54 @@ type channelMixin struct {
 	StableChannel    bool `long:"stable" description:"Install from the stable channel"`
 }
 
-func (x *channelMixin) setChannelFromCommandline() error {
+func (mx *channelMixin) setChannelFromCommandline() error {
```
Comment (mvo5) [inline cmd/snap/cmd_snap_op.go]: I like `mx` :)

File: cmd/snap/cmd_snap_op.go [hunk 3]
```diff
@@ -298,9 +324,9 @@ func refreshAll() error {
 	return listSnaps(names)
 }
 
-func refreshOne(name, channel string) error {
+func refreshOne(name string, opts *client.SnapOptions) error {
```
Comment (mvo5) [inline cmd/snap/cmd_snap_op.go]: Nice!

File: cmd/snap/cmd_snap_op_test.go [hunk 4]
```diff
@@ -301,6 +301,95 @@ foo +4.2update1 +17 +bar +-.*
...
+	_, err := snap.Parser().ParseArgs([]string{"refresh", "--beta"})
+	c.Assert(err, check.ErrorMatches, `a snap name is needed to specify mode or channel flags`)
+}
+
+func (s *SnapOpSuite) TestRefreshAllModeFlags(c *check.C) {
+	s.RedirectClientToTestServer(nil)
+	_, err := snap.Parser().ParseArgs([]string{"refresh", "--devmode"})
+	c.Assert(err, check.ErrorMatches, `a snap name is needed to specify mode or channel flags`)
+}
+
```
Comment (mvo5) [inline cmd/snap/cmd_snap_op_test.go]: Nice set of tests!

**Comment by mvo5** [comment]
This looks good, a spread test would be nice indeed.

File: daemon/api_test.go [hunk 5]
```diff
@@ -1705,7 +1705,9 @@ func (s *apiSuite) TestRefresh(c *check.C) {
 	snapstateUpdate = func(s *state.State, name, channel string, userID int, flags snapstate.Flags) (*state.TaskSet, error) {
 		calledFlags = flags
 		calledUserID = userID
-		installQueue = append(installQueue, name)
+		if flags.DevModeAllowed() {
```
Comment (pedronis) [inline daemon/api_test.go]: this tweak of the test feels a bit random/arbitrary, why not two tests or recording this effect differently?

Comment (chipaca) [inline daemon/api_test.go]: mostly because of the amount of duplicated setup boilerplate, but yeah, it'd probably be better split

**Comment by pedronis** [comment]
+1 (notwithstanding that we really need to do the right thing(tm) for refresh-all at some point soon)

**Pull request merged**